File: .hermes/team/control_plane/stacks/backend/python-fastapi/templates/service.py
from typing import List, Optional

from .repository import ${ClassName}Repository
from .schemas import ${ClassName}DTO


class ${ClassName}Service:
    def __init__(self, repo: ${ClassName}Repository):
        self.repo = repo

    async def list(self) -> List[${ClassName}DTO]:
        return await self.repo.list()

    async def get_by_id(self, item_id: int) -> Optional[${ClassName}DTO]:
        return await self.repo.get_by_id(item_id)

    async def create(self, dto: ${ClassName}DTO) -> ${ClassName}DTO:
        return await self.repo.create(dto)

    async def update(self, item_id: int, dto: ${ClassName}DTO) -> ${ClassName}DTO:
        return await self.repo.update(item_id, dto)

    async def delete(self, item_id: int) -> None:
        await self.repo.delete(item_id)

File: .hermes/team/control_plane/stacks/backend/go-gin/templates/service.go
package service

import (
	"errors"
)

type ${ClassName}Service struct {
	repo *${ClassName}Repository
}

func New${ClassName}Service(repo *${ClassName}Repository) *${ClassName}Service {
	return &${ClassName}Service{repo: repo}
}

func (s *${ClassName}Service) List() ([]${ClassName}DTO, error) {
	return s.repo.List()
}

func (s *${ClassName}Service) GetByID(id int64) (${ClassName}DTO, error) {
	item, err := s.repo.GetByID(id)
	if err != nil {
		return ${ClassName}DTO{}, errors.New("not found")
	}
	return item, nil
}

func (s *${ClassName}Service) Create(dto ${ClassName}DTO) (${ClassName}DTO, error) {
	return s.repo.Create(dto)
}

func (s *${ClassName}Service) Update(id int64, dto ${ClassName}DTO) (${ClassName}DTO, error) {
	dto.ID = id
	return s.repo.Update(dto)
}

func (s *${ClassName}Service) Delete(id int64) error {
	return s.repo.Delete(id)
}
